server.js

```diff
@@ -1,32 +1,153 @@
+// if you create an endpoint that takes params , you have to include that 
+// param in your endpoint. Otherwise it's not going to trigger that endpoint.
+// Query params are optional, you can include them, but you dont have to
+// the endpint is going to work with or without query in end point path
+
+// path params uses when you want only send one object from the data
+// query params uses when you about filtering a data and send back multiple amount, 
+// just a smaller array of the records 
+
+// FIRST ENDPOINT 
+// call back function inside of get() communicates with fronted via those two argument(req and res)
+// res = is what we send back to the frontend 
+// req = it's handaling send to the backend like a specific query for ex 
+app.get('/shows', (req, res) => {
+  // req is an object with properties such as query and params, 
+  // both query and params are an empty object from the begging 
+  // every query parameter consist of two part, key and value, like an object
+  const { title, country, type, year, page, limit } = req.query
+
+  // creating this variable, because we will always return the same array to the user
+  // note to me: it is my a currentQuestion variable from previous project?
+  let arrayToSendToUser = netflixData
+
+  // can i combine title, country and type into one if?
+  if (title) {
+    /* toLowerCase() sets the data in type array data to lowercase, 
+    then includes() returns true if a string contains type from url, 
+    includes() is almost the same as indexOf(), indecOf return number, when includes a boolean, 
+    a part of the string that we send should be in the database */
+    arrayToSendToUser = arrayToSendToUser.filter(
+      (item) => item.title.toLowerCase().includes(title.toLowerCase()) 
+    )
+  }
+  console.log(title)
+
+  if (country) {
+    arrayToSendToUser = arrayToSendToUser.filter(
+      (item) => item.country.toLowerCase().includes(country.toLowerCase()) 
+    )
+  }
+ 
+  if (type) {
+    arrayToSendToUser = arrayToSendToUser.filter(
+      (item) => item.type.toLowerCase().includes(type.toLowerCase()) 
+    )
+  }
+
+  if (year) {
+    // "+" turns the string in to a number.
+    arrayToSendToUser = arrayToSendToUser.filter((item) => item.release_year === +year)
+  }
+
+  // PAGINATION
+  // with limit user setting the limit of items to show per page 
+  // and page shows first items depends on what limit is
+  if (page && limit) {
+    // -1 because index starts with 0, multiply 0 so we start at 0
+    const startIndex = (page - 1) * limit
+    // here we get the end of an array
+    const endIndex = page * limit 
+    // slice gives us what in between startIndex and endIndex
+    const paginationArray = arrayToSendToUser.slice(startIndex, endIndex)
+    // returs an array with pagination
+    res.json(paginationArray)
+  }
+
+  // ERRORS
+  if (arrayToSendToUser.length === 0) {
+    res.status(404).json({
+      response: "This array is empty",
+      success: false
+    })
```

Comment:
I can just agree with Rosanna, amazing code, good descriptions, pagenation and the frontend, wow! :D to go back to our discussion on if/else for the queary(when you helped me) I think this can solve errors on queary (arrayToSendToUser.length === 0 ||  !arrayToSendToUser ) or this // ERRORS
  if (arrayToSendToUser.length === 0) {
    res.status(404).json({
      response: [],
      success: false
    })
  } else if (!arrayToSendToUser) {
    res.status(404).json({
      response: "Not found",
      success: false
    })
  } 

Anyways what a great project and code!!!! :D